engine/crates/engine-v2/src/request/bind.rs

```diff
@@ -35,148 +64,341 @@ impl From<BindError> for GraphqlError {
+                Some((id, _)) => *id,
+                None => {
+                    // A bound fragment definition has no selection set, it was already bound. We
+                    // only keep it for errors (name/name_pos) and directives.
+                    let fragment_definition = BoundFragmentDefinition {
+                        name: name.to_string(),
+                        name_location: fragment_definition_pos,
+                        type_condition,
+                        directives: vec![],
+                    };
+                    let id = BoundFragmentDefinitionId::from(self.fragment_definitions.len());
+                    self.fragment_definitions
+                        .insert(name.to_string(), (id, fragment_definition));
+
+                    id
+                }
+            },
+        }))
+    }
+
+    fn bind_inline_fragment(
+        &mut self,
+        parent: ParentType,
+        Positioned {
+            pos: location,
+            node: fragment,
+        }: Positioned<engine_parser::types::InlineFragment>,
+    ) -> BindResult<BoundSelection> {
+        let type_condition = fragment
+            .type_condition
+            .map(|condition| self.bind_type_condition(parent, &condition))
+            .transpose()?;
+        let fragment_root = type_condition.map(Into::into).unwrap_or(parent);
+        let selection_set_id = self.bind_selection_set(fragment_root, fragment.selection_set)?;
+        Ok(BoundSelection::InlineFragment(BoundInlineFragment {
+            location,
+            type_condition,
+            selection_set_id,
+            directives: vec![],
+        }))
+    }
+
+    fn bind_type_condition(
+        &self,
+        parent: ParentType,
+        Positioned { pos: location, node }: &Positioned<engine_parser::types::TypeCondition>,
+    ) -> BindResult<TypeCondition> {
+        let location = *location;
+        let name = node.on.node.as_str();
+        let definition = self
+            .schema
+            .definition_by_name(name)
+            .ok_or_else(|| BindError::UnknownType {
+                name: name.to_string(),
+                location,
+            })?;
+        let type_condition = match definition {
+            Definition::Object(object_id) => TypeCondition::Object(object_id),
+            Definition::Interface(interface_id) => TypeCondition::Interface(interface_id),
+            Definition::Union(union_id) => TypeCondition::Union(union_id),
+            _ => {
+                return Err(BindError::InvalidTypeConditionTargetType {
+                    name: name.to_string(),
+                    location,
+                });
+            }
+        };
+        let possible_types = TypeCondition::from(parent)
+            .resolve(self.schema)
+            .possible_types()
+            .iter()
+            .copied()
+            .collect::<HashSet<_>>();
+        let frament_possible_types = type_condition
```

Comment:
typo but no need to go through a whole CI cycle to fix it

Reply:
ah good catch!